src/main/java/org/opensearch/geospatial/index/mapper/xypoint/XYPointIndexer.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.geospatial.index.mapper.xypoint;
+
+import static org.opensearch.geospatial.index.common.xyshape.XYShapeConverter.toXYPoint;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import lombok.AllArgsConstructor;
+
+import org.apache.lucene.document.XYPointField;
+import org.apache.lucene.geo.XYPoint;
+import org.apache.lucene.index.IndexableField;
+import org.opensearch.geometry.Point;
+import org.opensearch.index.mapper.AbstractGeometryFieldMapper;
+import org.opensearch.index.mapper.ParseContext;
+
+/**
+ * Converts points into Lucene-compatible form for indexing in a xy_point field.
+ */
+@AllArgsConstructor
+public class XYPointIndexer implements AbstractGeometryFieldMapper.Indexer<List<ParsedXYPoint>, List<? extends XYPoint>> {
```

Comment:
why we are not implementing : AbstractPointGeometryFieldMapper?

Reply:
Indexer interface is available only in AbstractGeometryFieldMapper class

Reply:
@VijayanB can please check this. I think you worked on the same for the ShapeIndexer, I wasn't able to remember what we discussed here last time

Reply:
Indexer should implement AbstractGeometryFieldMapper.Indexer. XYPointFieldMapper will implement AbstractGeometryFieldMapper